iroh/src/node.rs

```diff
@@ -106,10 +104,12 @@ pub type IrohServerEndpoint = quic_rpc::transport::boxed::ServerEndpoint<
 #[derive(Debug, Clone)]
 pub struct Node<D> {
     inner: Arc<NodeInner<D>>,
-    task: SharedAbortingJoinHandle<()>,
+    task: Shared<MapErr<AbortOnDropHandle<()>, JoinErrToStr>>,
```

Comment:
Maybe its worth adding a comment here why this construction exists and what magic is exactly performs.  Right now I know, but I can imagine being baffled sometime in the future.

Reply:
Agreed, will add that :)